tsdb/engine/tsm1/wal.go

```diff
@@ -27,6 +27,9 @@ const (
 
 	defaultBufLen = 1024 << 10 // 1MB (sized for batches of 5000 points)
 
+	// walEncodeBufSize is the size of the wal entry encoding buffer
+	walEncodeBufSize = 4 * 1024 * 1024
```

Comment:
does this get allocated for every shard? This going to be a problem on systems with hundreds or thousands of shards?